### DISCUSSION TITLE
Packet Loss Testing

### DISCUSSION THREAD

**Discussion by UncommonGarb** [alexjustesen/speedtest-tracker#327] [Ideas]
In my head this seems like an out of scope request, but it would add to an all in one solution. 

Packet loss testing, periodic packet loss testing, and results shown along side speed speedtests as well as in a a graph of success percentage during test.

**Comment by alexjustesen** [comment]
When we're talking packet loss are we looking specifically at the speedtest that are run or as a separate metric?

**Reply by UncommonGarb** [reply]
I am thinking as a completely separate metric and graph possibly using the same cron schedule and it run after. The graph would ultimately show a percentage graph over time like speed tests, but maybe more beneficial as a bar graph. 
2% packet loss - 8PM
20% packet loss - 9PM
![image](https://user-images.githubusercontent.com/61894631/212356577-738f1cae-f639-422f-b0e2-01d1e0e56cb2.png)

**Reply by alexjustesen** [reply]
So packet loss exists in the `data` column currently. Currently the InfluxDB integration has packet loss in the [payload](https://github.com/alexjustesen/speedtest-tracker/blob/55d65c840b16704dbfa04d2cb994d24f28ee08a5/app/Models/Result.php#L92) (where applicable as not all speedtest servers report this metric) so if you're looking to dashboard that metric out I suggest using InfluxDB and Graphana.

**Reply by UncommonGarb** [reply]
Interesting, I currently am trying to incorporate this data into Graphana with with Influx but ran into a roadblock not related to your app. I will continue to try and see where I get. Thank you for this information.

**Reply by alexdelprete** [reply]
> currently am trying to incorporate this data into Graphana with with Influx but ran into a roadblock not related to your app

If needed, I made a dashboard for ST2 that anybody can use and works out-of-the-box: https://grafana.com/grafana/dashboards/17460-speedtest-tracker-v2/

**Reply by UncommonGarb** [reply]
Got all this working, thank you for the direction.

**Reply by UncommonGarb** [reply]
I am just starting with influx and there is no historic data. Any chance there is away to mass import / export from speedtest-tracker and all of its history?

**Reply by alexdelprete** [reply]
You mean from the old ST project or the data of ST2? It's a good idea actually, if you're referring to the latter.

**Reply by UncommonGarb** [reply]
ghcr.io/alexjustesen/speedtest-tracker:latest

This is the docker image im using. trying to export out of this internal database to influx.

**Reply by UncommonGarb** [reply]
Currently I see no ability to force export historic data.

**Reply by alexdelprete** [reply]
There's not. The import from the old ST project (henrywhitaker3) is in the roadmap, but I think your idea to import data from ST2 when influx is configured after ST2 has already collected data should be implemented.

Can you please open a FR specifically for that? Thanks.

**Reply by UncommonGarb** [reply]
Okay I believe I am on ST2. I will open FR.

**Reply by alexdelprete** [reply]
You are in ST2, this is the repo of ST2, I call it ST2 to distinguish it from the old abandoned project, but the maintainer/dev @alexjustesen decided to keep the same name. I would've preferred to add a 2 to avoid confusion. :)
